cpp/tests/community/mg_ecg_test.cpp

```diff
@@ -0,0 +1,271 @@
+// Parameterized test fixture, to be used with TEST_P().  This defines common
+// setup and teardown steps as well as common utilities used by each E2E MG
+// test.  In this case, each test is identical except for the inputs and
+// expected outputs, so the entire test is defined in the run_test() method.
+//
+template <typename input_usecase_t>
+class Tests_MGEcg : public ::testing::TestWithParam<std::tuple<Ecg_Usecase, input_usecase_t>> {
+ public:
+  static void SetUpTestCase() { handle_ = cugraph::test::initialize_mg_handle(); }
+
+  static void TearDownTestCase() { handle_.reset(); }
+
+  // Run once for each test instance
+  virtual void SetUp() {}
+  virtual void TearDown() {}
+
+  template <typename vertex_t, typename edge_t, typename weight_t>
+  void compare_sg_results(
+    raft::handle_t const& handle,
+    raft::random::RngState& rng_state,
+    cugraph::graph_view_t<vertex_t, edge_t, false, true> const& mg_graph_view,
+    std::optional<cugraph::edge_property_view_t<edge_t, weight_t const*>> mg_edge_weight_view,
+    rmm::device_uvector<vertex_t> const& cluster_ids,
+    weight_t min_weight,
+    size_t ensemble_size,
+    size_t max_level,
+    weight_t threshold,
+    weight_t resolution,
+    weight_t mg_modularity)
+  {
+    auto& comm           = handle.get_comms();
+    auto const comm_rank = comm.get_rank();
+
+    cugraph::graph_t<vertex_t, edge_t, false, false> sg_graph(handle);
+    std::optional<
+      cugraph::edge_property_t<cugraph::graph_view_t<vertex_t, edge_t, false, false>, weight_t>>
+      sg_edge_weights{std::nullopt};
+    std::tie(sg_graph, sg_edge_weights, std::ignore) = cugraph::test::mg_graph_to_sg_graph(
+      *handle_,
+      mg_graph_view,
+      mg_edge_weight_view,
+      std::optional<raft::device_span<vertex_t const>>{std::nullopt},
+      false);  // crate an SG graph with MG graph vertex IDs
+
+    weight_t sg_modularity{-1.0};
+
+    if (comm_rank == 0) {
+      auto sg_graph_view = sg_graph.view();
+      auto sg_edge_weight_view =
+        sg_edge_weights ? std::make_optional((*sg_edge_weights).view()) : std::nullopt;
+
+      std::tie(std::ignore, std::ignore, sg_modularity) = cugraph::ecg(handle,
+                                                                       rng_state,
+                                                                       sg_graph_view,
+                                                                       sg_edge_weight_view,
+                                                                       min_weight,
+                                                                       ensemble_size,
+                                                                       max_level,
+                                                                       threshold,
+                                                                       resolution);
+    }
+    if (comm_rank == 0) {
+      // FIXME: SG and MG modularity values can differ by a few percent,
```

Comment:
Do you have a strategy for comparing results?  We should also test SG, which I don't see yet in the PR.